### ISSUE TITLE
Python 3.6.0 Support

### ISSUE THREAD

**Issue by Siecje** [jamesabel/osnap#20]
Here https://github.com/jamesabel/osnap/commit/2e4a7f7c516a3b38496f7d65bd871efaed8858eb#diff-e272fd5500f2ed699f985d24b5403b06R10 you say that you can't install `pip` when using the embeddable zip file version of Python 3.6.0.

How are you installing pip?

I tried both `python.exe -m ensurepip` and `python.exe -m pip install pip` and neither of them work in Python 3.6.0 or Python 3.5.2

**Comment by jamesabel** [comment]
Since embeddable Python doesn't include pip I get it from [https://bootstrap.pypa.io/get-pip.py](https://bootstrap.pypa.io/get-pip.py) and just `python.exe get-pip.py` .  Unfortunately on 3.6.0 I encounter this bug: [https://github.com/pypa/pip/issues/4207](https://github.com/pypa/pip/issues/4207) .

**Comment by Siecje** [comment]
Which Python do you use on macOS?

**Comment by jamesabel** [comment]
On MacOS I use 3.6.0.

**Comment by Siecje** [comment]
But there is no embeddable version for macOS

**Comment by jamesabel** [comment]
That's basically correct - there is no embeddable Python for MacOS from python.org (eGenix has [PyRun](http://www.egenix.com/products/python/PyRun/) but I've had some issues with that).  That's why in osnap I build Python on the target machine, which in my mind is a workaround until we can get the mainstream freezers like py2app or briefcase to work with the difficult-to-freeze cases.  BTW I created [spafit](https://github.com/jamesabel/spafit) as a difficult-to-freeze test case.

**Comment by jamesabel** [comment]
BTW, do you have an open source app that you freeze/install?  If so, can you give me a pointer?  I might be collecting some real-world apps for freezer/installer test cases.

**Comment by Siecje** [comment]
The application I distribute is not open source but I have a simple PyQt5 QML Application here https://github.com/siecje/qml-testing

The branches are attempts at using the different freezers.

**Comment by jamesabel** [comment]
For each freezer you've tried, can you tell me if it worked or not?  And if it didn't work, can you provide the error message?  I'm also trying to collect people's experiences with freezers.

**Comment by Siecje** [comment]
cx_Freeze, pynsist, osnap worked

PyInstaller and pyqtdeploy did not work

The open issues here show the errors https://github.com/Siecje/qml-testing/issues

**Comment by jamesabel** [comment]
Thanks - just a couple more questions:

Are you Windows only or do you also distribute for MacOS?

Since multiple worked, which freezer do you actually distribute with?

**Comment by Siecje** [comment]
I would like to distribute for macOS and Windows but I don't have a Mac to create the installer.

**Comment by jamesabel** [comment]
For the Windows version which tool did you end up using?

**Comment by Siecje** [comment]
cx_Freeze but with some problems.

I need to try `osnap` again.

`pynsist` works and is consistently reproducible and easily reproducible by other people (unlike cx_Freeze) but it exposes the source code.

**Comment by jamesabel** [comment]
For Windows, `osnap` probably isn't going to get you much over `pynsist`, but of course you're welcome to use it!

Protecting "Intellectual Property" - and what is 'good enough' protection -  is another matter altogether.  :)

**Comment by Siecje** [comment]
But you don't distribute .py files do you?

I just tried `osnap` and I get the Python has stopped working on Windows 10 with this app

https://github.com/Siecje/qml-testing/tree/osnap

**Comment by jamesabel** [comment]
.py files are included in the dist, since everything is included (it's simplistic ...).

On your issue above, I think it's that you are specifying an old PyQt5.  I forked it and the code here worked for me: https://github.com/jamesabel/qml-testing/tree/osnap .  But if I used PyQt5==5.7 as you specify in your requirements.txt, it crashed python itself even without osnap (just creating a normal venv). 

Here's the installer that I created (it works):
https://dl.dropboxusercontent.com/u/3154988/app_installer.exe
